### PR TITLE
glpaper: init at unstable-2020-03-30

### PR DESCRIPTION
**Pull request by ccellado**
###### Motivation for this change

Add new useful package to the Wayland stack.

###### Things done

Added glpaper package.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/glpaper/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  version = "unstable-2020-03-30";
+
+  src = fetchhg {
+    url = "https://hg.sr.ht/~scoopta/glpaper";
+    rev = "a95db77088dfb5636a87f3743fc9b5eca70c1ae2";
+    sha256 = "04y12910wvhy4aqx2sa63dy9l6nbs7b77yqpdhc96x2b3mgzgjfs";
+  };
+
+  nativeBuildInputs = [ meson ninja pkg-config ];
+  buildInputs = [ wayland wayland-protocols wlroots libX11 libGL ];
```
Comment (jtojnar) [inline pkgs/development/tools/glpaper/default.nix:15]: Could you add a comment to the dependencies not listed in the `meson.build` file why they are needed?

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -1280,6 +1280,12 @@
     githubId = 64804;
     name = "Dennis Gosnell";
   };
+  ccellado = {
```
Comment (jtojnar) [inline maintainers/maintainer-list.nix:1283]: Could you make this a separate commit?

File: pkgs/development/tools/glpaper/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with stdenv.lib; {
+    description =
+      "Wallpaper program for wlroots based Wayland compositors such as sway that allows you to render glsl shaders as your wallpaper";
+    homepage = "https://hg.sr.ht/~scoopta/glpaper";
+    license = licenses.gpl3Plus;
+    platforms   = platforms.linux;
```
Comment (jtojnar) [inline pkgs/development/tools/glpaper/default.nix:22]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/development/tools/glpaper/default.nix (hunk 1 above)
Comment (ccellado) [reply pkgs/development/tools/glpaper/default.nix:15]: By removing non-meson deps one by one this is the list. 
Compilation failed otherwise.

How should i comment this then?

Comment (jtojnar) [reply pkgs/development/tools/glpaper/default.nix:15]: wlroots or wayland-protocols do not seem to be required

```suggestion
  buildInputs = [
    wayland
    libX11 # required by libglvnd
    libGL
  ];
```

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (jtojnar) [reply maintainers/maintainer-list.nix:1283]: the commit should be named `maintainers: add ccellado`, see
https://github.com/ccellado/nixpkgs/commits/glpaper-unstable/maintainers/maintainer-list.nix

Comment (ccellado) [reply maintainers/maintainer-list.nix:1283]: Changed

**Comment by jtojnar** [comment]
Could you also rebase the branch to fixup `Update pkgs/development/tools/glpaper/default.nix` onto the glpaper commit?

**Comment by ccellado** [comment]
> Could you also rebase the branch to fixup `Update pkgs/development/tools/glpaper/default.nix` onto the glpaper commit?

Squashed, did that

**Comment by jtojnar** [comment]
Could you also move the maintainer commit before the package commit (e.g. using `git rebase -i HEAD^^`). Otherwise there will be broken evaluation which might break `git bisect`.

**Comment by ccellado** [comment]
> Could you also move the maintainer commit before the package commit (e.g. using `git rebase -i HEAD^^`). Otherwise there will be broken evaluation which might break `git bisect`.

Looks like it worked.

**Comment by jtojnar** [comment]
Could you please push the changes?

**Comment by ccellado** [comment]
> Could you please push the changes?

I don't quite understand. I've pushed after reordering, my commit list here displays Maintainer commit before the main commit.

**Comment by jtojnar** [comment]
Hmm, you are right, GitHub just displays them in the wrong order here:

image

**Comment by jtojnar** [comment]
Thanks.

**Comment by ccellado** [comment]
> Thanks.

Thank your patience, I'm new to contributing <3
